tests/e2e/common/diagnostics/extensions.go

```diff
@@ -0,0 +1,60 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package diagnostics
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+type nodeFeatures struct {
+	*Config
+}
+
+type nodeFeatureRules struct {
+	*Config
+}
+
+func (c nodeFeatures) Collect(ctx context.Context) error {
+	nfs, err := c.NfdClient.NfdV1alpha1().NodeFeatures(c.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.ErrorS(err, "Error getting node features")
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("error collecting %t: %w", c, err)
```
This should ensure that we log the type and the error that was raised.

Reply:
This probably holds for all other `Collect` implementations apart from the top-level one.